src/main/java/dev/latvian/mods/kubejs/neoforge/NeoForgeEventsLookup.java

```diff
@@ -0,0 +1,206 @@
+					continue;
+				}
+
+				var eventClass = getClass(className);
+				if (eventClass == null) {
+					continue;
+				}
+
+				var formattedName = formatEventName(className.substring(className.lastIndexOf('.') + 1));
+				if (eventMap.containsKey(formattedName)) {
+					KubeJS.LOGGER.warn("[NativeEvents] Event class " + className + " for name '" + formattedName + "' is already registered as: " + eventMap.get(formattedName));
+					continue;
+				}
+
+				eventMap.put(formattedName, eventClass);
+			}
+		}
+
+		KubeJS.LOGGER.info("[NativeEvents] Loaded " + eventMap.size() + " events");
+	}
+
+	/**
+	 * Get all invalid classes which should not be loaded.
+	 * This includes all classes which are annotated with {@link Mixin} or with {@link OnlyIn}, when they current dist does not match.
+	 *
+	 * @param annotations all annotations
+	 * @return set of invalid classes
+	 */
+	private Set<String> getInvalidClasses(Set<ModFileScanData.AnnotationData> annotations) {
+		Type onlyIn = Type.getType(OnlyIn.class);
+		Type mixin = Type.getType(Mixin.class);
+
+		Set<String> invalid = new HashSet<>();
+		for (ModFileScanData.AnnotationData annotation : annotations) {
+			if (!annotation.targetType().equals(ElementType.TYPE)) {
+				continue;
+			}
+
+			if (isInvalidDist(annotation, onlyIn) || isMixin(annotation, mixin)) {
+				invalid.add(annotation.clazz().getClassName());
+			}
+		}
+
+		return invalid;
+	}
+
+	/**
+	 * Checks if given className is part of invalid classes. It also checks if the super class is part of invalid classes.
+	 *
+	 * @param invalidClasses set of invalid classes
+	 * @param className      class name
+	 * @return true if invalid
+	 */
+	private boolean isInvalidClass(Set<String> invalidClasses, String className) {
+		if (invalidClasses.contains(className)) {
+			return true;
+		}
+
+		String[] parts = className.split("\\$", 2);
+		return invalidClasses.contains(parts[0]);
+	}
+
+	private boolean isMixin(ModFileScanData.AnnotationData annotation, Type mixinType) {
+		return annotation.annotationType().equals(mixinType);
+	}
+
+	private boolean isInvalidDist(ModFileScanData.AnnotationData annotation, Type distType) {
+		Type type = annotation.annotationType();
+		if (!type.equals(distType)) {
+			return false;
+		}
+
+		Object unknownValue = annotation.annotationData().get("value");
+		if (unknownValue instanceof ModAnnotation.EnumHolder enumHolder) {
+			String invalidValue = FMLEnvironment.dist == Dist.CLIENT ? Dist.DEDICATED_SERVER.name() : Dist.CLIENT.name();
+			return enumHolder.getValue().equals(invalidValue);
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks if given class name is invalid. Only classes which end with "Event" are valid, but also subtypes of it.
+	 *
+	 * @param fullClassName class name
+	 * @return true if class name is invalid
+	 */
+	private boolean isInvalidClassName(String fullClassName) {
+		String[] parts = fullClassName.split("\\$");
+		for (String part : parts) {
+			if (part.endsWith("Event")) {
```

Comment:
I don't like this as a way to check something is an event. There is no garuntee that some idiot   isn't going to end their custom events with Event (ie I could see someone making a ModIDCustomEvents class that holds a bunch of Event inner classes, excpet they are named stuff like GrowingPlant, ending up with stuff like `ModIDCustomEvents$GrowingPlant` as a plant growing event).

Reply:
As this is only used for the custom calls, I only include neoforge own events and they at least follow this rule. Modded events should still go to `onEvent` imo.

Reply:
Ah, okay. Thats good.
I wasn't sure if this was for all events or for just the 'short' version.